source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -87,8 +87,79 @@ FileImplWin32::FlagsAndMode FileImplWin32::translateFlag(FlagSet in) {
   return {access, creation};
 }
 
+Api::IoCallBoolResult ConsoleFileImplWin32::open(FlagSet) {
+  FlagsAndMode flags;
+  flags.access_ = GENERIC_WRITE | GENERIC_READ;
+  flags.creation_ = OPEN_EXISTING;
+  fd_ = CreateFileA(path().c_str(), flags.access_, FILE_SHARE_READ | FILE_SHARE_WRITE, 0,
+                    flags.creation_, 0, NULL);
+  if (fd_ == INVALID_HANDLE) {
+    return resultFailure(false, ::GetLastError());
+  }
+  return resultSuccess(true);
+}
+
+Api::IoCallBoolResult StdOutFileImplWin32::open(FlagSet) {
+  fd_ = GetStdHandle(std_handle_);
+  if (fd_ == NULL) {
+    // If an application does not have associated standard handles,
+    // such as a service running on an interactive desktop
+    // and has not redirected them, the return value is NULL.
+    // In that case we throw an exception instead of failing since there is no last error.
+    throw EnvoyException(
+        fmt::format("The process does not have an associated handle for `stdout`"));
```

Comment:
Is it appropriate to throw in this case?  I'm not sure that this open call always happens within a try/catch block, should this be treated the same as failure to open a file due to invalid filesystem path?

Reply:
The only caller that I could find in the codebase for `Filesystem::Open()` is from `AccessLogFileImpl::open()` which then turns it into an exception. This is why I thought it would be ok to throw the exception there

Reply:
I would suggest returning an error and allowing the caller to continue converting to exception.

Reply:
I changed in the implementation in 885899b4d25610001db3998d598c0dcd0f8fb2c8